### ISSUE TITLE
Find fix for pet movement issues

### ISSUE THREAD

**Issue by Chris6ix** [datatags/UltraCosmetics#23]
### Describe the feature and how it would be used

Just using this issue to keep track of all the pets that needed to be disabled because of movement issues, so that we can try to find fixes for them.
The specific reasons for disabling each pet can be found in https://github.com/Chris6ix/UltraCosmetics/blob/Chris6ix-more-pets/core/src/main/java/be/isach/ultracosmetics/cosmetics/type/PetType.java

Currently disabled pets:
- Slime
- Vex
- All the pets in #22 that have a "Disabled" comment next to them

### Are there alternatives?

/

### Additional information

/

**Comment by datatags** [comment]
In UC v3, I'm hoping to drop pet support from legacy versions and switch to MobChip for pet AI, so that may fix some of these issues. At the very least, I'll be able to talk about it with people who are familiar with how path finding works

**Comment by datatags** [comment]
Vex pet now works as of dd792a4d 😄

**Comment by Chris6ix** [comment]
> 

Nice!
I'll remove it from the list then

**Comment by dennishzg** [comment]
Looks like the Squid and Glow Squid are working now. They teleport to the player, like the Pufferfish, Cod etc.

**Comment by datatags** [comment]
> Looks like the Squid and Glow Squid are working now. They teleport to the player, like the Pufferfish, Cod etc.
> 
> Tested with: [bc44b67](https://github.com/datatags/UltraCosmetics/commit/bc44b67264cfbb0659e625d3fb7496328109b396) and `Paper version git-Paper-345 (MC: 1.19.3) (Implementing API version 1.19.3-R0.1-SNAPSHOT) (Git: e498dda)`.

Yeah but the goal is for squid and glow squid to be able to float toward the player in water, not just teleport. I still need to look into why that is.

Slime and magma cube now work, so I've taken those off the list.
